### DISCUSSION TITLE
Why minitest.py is failing for majority of (sigs,kex) pairs ?

### DISCUSSION THREAD

**Discussion by gobbledy-gook** [open-quantum-safe/oqs-provider#507] [Q&A]
I locally setup `minitest.py`, `liboqs-python` and got a successful connection on port 6054 with `test.openquantumsafe.org` but I am unable to make a successful connection using sig:dilithium2 and kex:kyber512.

I am not sure if it is supposed to fail or if am I missing something.

![Screenshot 2024-09-01 at 12 03 14 PM](https://github.com/user-attachments/assets/4d36246a-0fc9-4b68-8076-d19e94b7fea4)

Any solutions or suggestions would be of great help!
Thanks!

**Comment by baentsch** [comment]
> got a successful connection on port 6054 with test.openquantumsafe.org

That is not tell-tale as that port does not use PQC.

> I am not sure if it is supposed to fail or if am I missing something.

Please state exactly which algorithm combinations (ports) do and don't work. A typical error is that one uses an `openssl` version (baked into `python`) that [does not yet support provider-based signatures](https://github.com/open-quantum-safe/oqs-provider?tab=readme-ov-file#note-on-openssl-versions): Be sure to have `openssl version` state 3.2.0 or higher. KEMs (combined with classic signatures like EC or RSA, i.e. all ports 6000-6108 at test.openquantumsafe.org) should always work.

**Comment by baentsch** [comment]
In looking through unanswered discussions, I stumbled on this -- I guess this is to be closed now, right @gobbledy-gook ?

**Reply by gobbledy-gook** [reply]
yes indeed!
